autopr/services/generation_service.py

```diff
@@ -77,7 +77,11 @@ def _repo_to_files_and_token_lengths(
                 continue
             if excluded_files is not None and blob.path in excluded_files:
                 continue
-            content = blob.data_stream.read().decode()
+            try:
+                content = blob.data_stream.read().decode()
+            except UnicodeDecodeError:
+                log.debug(f"Error decoding file: {blob.path}")
+                continue
```

Comment:
Hey, thanks so much for the PR!

This whole method `_repo_to_files_and_token_lengths` is obsolete actually, a relic of the past. Could you remove it altogether?

Reply:
Thanks for the heads up! I'll go ahead and give that old relic the boot. Let me know if there's anything else you'd like me to do. I'm all ears!